File: .claude/skills/fixtures/core6-mechanical/typescript/inline/before.ts
function formatUserEmail(email: string): string {
  return email.trim().toLowerCase();
}

function createUserRecord(name: string, email: string): Record<string, string> {
  const normalizedEmail = formatUserEmail(email);
  return { name, email: normalizedEmail };
}

const record = createUserRecord('Alice', '  Alice@Example.COM  ');
console.log(record);

File: .claude/skills/fixtures/core6-mechanical/rust/inline/before.rs
fn normalize_email(email: &str) -> String {
    email.trim().to_lowercase()
}

fn create_user(name: &str, email: &str) -> (String, String) {
    let normalized_email = normalize_email(email);
    (name.to_string(), normalized_email)
}

fn main() {
    let (name, email) = create_user("Alice", "  Alice@Example.COM  ");
    println!("User: {} <{}>", name, email);
}
